airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -289,8 +297,8 @@ def _fetch_saml_assertion_using_http_spegno_auth(self, saml_config: Dict[str, An
             raise ValueError('Invalid SAML Assertion')
         return saml_assertion
 
-    def _assume_role_with_web_identity(self, role_arn, assume_role_kwargs, base_session):
-        base_session = base_session or botocore.session.get_session()
+    def _get_web_identity_credential_fetcher(self):
```

Comment:
I know this didn't have one previously, but could you drop a return type on the signature?  I think it should be `botocore.credentials`?

Reply:
Done, thanks for the feedback.